robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowImageView.java

```diff
@@ -34,7 +34,11 @@ public Bitmap getImageBitmap() {
 
   @Implementation
   public void setImageDrawable(Drawable drawable) {
-    this.imageDrawable = drawable;
+
+    if(this.imageDrawable != drawable) {
+      this.imageDrawable = drawable;
+      invalidate();
```

Comment:
I want the shadow's wasInvalidated method to return true. Does that require calling the shadow's invalidate and relying on that to call the real method? If not, we could use the directlyOn call here.